pallets/ajuna-tournament/src/lib.rs

```diff
@@ -0,0 +1,753 @@
+					ensure!(percentage > 0, Error::<T, I>::InvalidTournamentConfig);
+					percentage
+				},
+			} as u16;
+
+			ensure!(
+				(reward_table_total_dist + golden_duck_dist) <= 100,
+				Error::<T, I>::InvalidTournamentConfig
+			);
+
+			ensure!(
+				config.max_players > 0 && config.max_players <= MAX_PLAYERS,
+				Error::<T, I>::InvalidTournamentConfig
+			);
+
+			Ok(())
+		}
+
+		fn try_insert_tournament_schedule(
+			season_id: &T::SeasonId,
+			tournament_id: &TournamentId,
+			config: &TournamentConfigFor<T, I>,
+		) -> DispatchResult {
+			TournamentSchedules::<T, I>::try_mutate(config.start, |action| match action {
+				None => {
+					*action = Some(TournamentScheduledAction::StartActivePhase(
+						*season_id,
+						*tournament_id,
+					));
+					Ok(())
+				},
+				Some(_) => Err(Error::<T, I>::CannotScheduleTournament),
+			})?;
+			TournamentSchedules::<T, I>::try_mutate(config.active_end, |action| match action {
+				None => {
+					*action = Some(TournamentScheduledAction::SwitchToClaimPhase(
+						*season_id,
+						*tournament_id,
+					));
+					Ok(())
+				},
+				Some(_) => Err(Error::<T, I>::CannotScheduleTournament),
+			})?;
+			TournamentSchedules::<T, I>::try_mutate(config.claim_end, |action| match action {
+				None => {
+					*action =
+						Some(TournamentScheduledAction::EndClaimPhase(*season_id, *tournament_id));
+					Ok(())
+				},
+				Some(_) => Err(Error::<T, I>::CannotScheduleTournament),
+			})?;
+
+			Ok(())
+		}
+
+		fn try_get_active_tournament_id_for(
+			season_id: &T::SeasonId,
+		) -> Result<TournamentId, DispatchError> {
+			match ActiveTournaments::<T, I>::get(season_id) {
+				TournamentState::Inactive => Err(Error::<T, I>::NoActiveTournamentForSeason.into()),
+				TournamentState::ActivePeriod(tournament_id) |
+				TournamentState::ClaimPeriod(tournament_id, _) => Ok(tournament_id),
+			}
+		}
+
+		/// Returns the available funds for payout in the winner/golden duck accounts.
+		/// If the amount is limited by 'max_reward' config in the tournament the amount will be
+		/// limited to that amount.
+		fn get_reward_payout(
+			tournament_config: &TournamentConfigFor<T, I>,
+			treasury_account: &AccountIdFor<T>,
+		) -> BalanceOf<T, I> {
+			let total_payout = T::Currency::free_balance(treasury_account);
+			if let Some(max_payout) = tournament_config.max_reward {
+				sp_std::cmp::min(total_payout, max_payout)
+			} else {
+				total_payout
+			}
+		}
+
+		fn try_update_rank_table(
+			table: &mut RankingTableFor<T, I>,
+			tournament_config: &TournamentConfigFor<T, I>,
+			index: usize,
+			entity_id: &T::EntityId,
+			entity: &T::RankedEntity,
+		) -> Option<bool> {
+			if index < tournament_config.max_players as usize {
+				if table.len() == tournament_config.max_players as usize {
+					let _ = table.pop();
+				}
+
+				if table.force_insert_keep_left(index, (entity_id.clone(), entity.clone())).is_ok()
+				{
+					Some(true)
+				} else {
+					Some(false)
+				}
+			} else {
+				None
```

Comment:
None basically means that the entity didn't enter the ranking table, it's not really an error since you can try to rank an entity that simply put doesn't make the cut, the only error here is the `Some(false)` because that means you should have inserted the entity in the table but you were not able.

Reply:
I see, and I think that your comments have greatly helped me to understand what happens. Personally, I think we can make the code even better to understand if we introduce types that carry such meaning directly, I prepared a suggestion in https://github.com/ajuna-network/ajuna-pallets/pull/21